diff --git a/components/Code.astro b/components/Code.astro
index 8bd3476a1133b515b68eab90fab078b3d5bece3b..15aa21e27b8962d10d4f70678c9bf8e75f19a477 100644
--- a/components/Code.astro
+++ b/components/Code.astro
@@ -37,7 +37,7 @@ const { code, lang = 'plaintext', theme = 'github-dark', wrap = false } = Astro.
 /** Replace the shiki class name with a custom astro class name. */
 function repairShikiTheme(html: string): string {
 	// Replace "shiki" class naming with "astro"
-	html = html.replace('<pre class="shiki"', '<pre class="astro-code"');
+	html = html.replace(/<pre class="(.*)shiki(.*)"/, '<pre class="$1astro-code$2"');
 	// Handle code wrapping
 	// if wrap=null, do nothing.
 	if (wrap === false) {
diff --git a/dist/core/build/vite-plugin-internals.js b/dist/core/build/vite-plugin-internals.js
index d6a1c2ca51c7a7bdc8d29246f696b1ea20eb87b8..c2911258e4c2aa01000e73894b222a8c81f5274c 100644
--- a/dist/core/build/vite-plugin-internals.js
+++ b/dist/core/build/vite-plugin-internals.js
@@ -29,7 +29,11 @@ function vitePluginInternals(input, internals) {
         promises.push(
           this.resolve(specifier).then((result) => {
             if (result) {
-              mapping.set(result.id, specifier);
+              if (mapping.has(result.id)) {
+                mapping.get(result.id).push(specifier);
+              } else {
+                mapping.set(result.id, [specifier]);
+              }
             }
           })
         );
@@ -37,8 +41,10 @@ function vitePluginInternals(input, internals) {
       await Promise.all(promises);
       for (const [, chunk] of Object.entries(bundle)) {
         if (chunk.type === "chunk" && chunk.facadeModuleId) {
-          const specifier = mapping.get(chunk.facadeModuleId) || chunk.facadeModuleId;
-          internals.entrySpecifierToBundleMap.set(specifier, chunk.fileName);
+          const specifiers = mapping.get(chunk.facadeModuleId) || [chunk.facadeModuleId];
+          specifiers.forEach(x => {
+            internals.entrySpecifierToBundleMap.set(x, chunk.fileName);
+          });
         } else if (chunk.type === "chunk") {
           for (const id of Object.keys(chunk.modules)) {
             const pageData = internals.pagesByViteID.get(id);