diff --git a/dist/core/build/vite-plugin-internals.js b/dist/core/build/vite-plugin-internals.js
index e03b64e8f3f61e85d9a14db927997f3357e3e197..0373d72c27eae0daa6b93075930416862b6ba3fb 100644
--- a/dist/core/build/vite-plugin-internals.js
+++ b/dist/core/build/vite-plugin-internals.js
@@ -29,7 +29,11 @@ function vitePluginInternals(input, internals) {
         promises.push(
           this.resolve(specifier).then((result) => {
             if (result) {
-              mapping.set(result.id, specifier);
+              if (mapping.has(result.id)) {
+                mapping.get(result.id).push(specifier);
+              } else {
+                mapping.set(result.id, [specifier]);
+              }
             }
           })
         );
@@ -37,8 +41,10 @@ function vitePluginInternals(input, internals) {
       await Promise.all(promises);
       for (const [, chunk] of Object.entries(bundle)) {
         if (chunk.type === "chunk" && chunk.facadeModuleId) {
-          const specifier = mapping.get(chunk.facadeModuleId) || chunk.facadeModuleId;
-          internals.entrySpecifierToBundleMap.set(specifier, chunk.fileName);
+          const specifiers = mapping.get(chunk.facadeModuleId) || [chunk.facadeModuleId];
+          specifiers.forEach(x => {
+            internals.entrySpecifierToBundleMap.set(x, chunk.fileName);
+          });
         }
       }
     }